wfc_dylib/src/lib.rs

```diff
@@ -444,11 +478,25 @@ pub extern "C" fn wfc_observe(
         &mut *wfc_rng_state_handle.0
     };
 
+    let observations = unsafe {
+        assert!(!spent_observations.is_null());
```

Comment:
I thought we wanted this optional? This is not what optional looks like,

Reply:
This is reporting the number of observations that had to be made. It is an important information even when the observations are not to be limited.

Reply:
Yes, but you told me either on call or on slack that this is going to be optional. I am just wondering what made you change your mind. I am not against it being mandatory.

Reply:
Don't remember that. Not to argue, but why would we make reporting of required observations optional? Does it cost anything?

Reply:
Just for more flexibility, but it doesn't matter much.